crates/volta-core/src/manifest/serial.rs

```diff
@@ -122,16 +123,35 @@ impl Manifest {
             }
         }
         Ok(manifest::Manifest {
-            platform: self.into_platform()?.map(Rc::new),
+            platform: self.to_platform()?.map(Rc::new),
             dependencies: self.dependencies,
             dev_dependencies: self.dev_dependencies,
             bin: map,
             engines: self.engines.map(|e| e.node),
         })
     }
 
-    pub fn into_platform(&self) -> Fallible<Option<platform::PlatformSpec>> {
-        if let Some(toolchain) = &self.toolchain {
+    pub fn to_platform(&self) -> Fallible<Option<platform::PlatformSpec>> {
+        // Backwards compatibility to allow users to upgrade to using the
+        // `volta` key simultaneously with the `toolchain` key, but with
+        // deprecation warnings about the use of `toolchain`. Prefer the `volta`
+        // key if it is set, and provide a deprecation warning if the user is
+        // using the `toolchain` key.
+        let (toolchain, deprecation) = match (&self.volta, &self.toolchain) {
+            (Some(volta), None) => (Some(volta), None),
+            (Some(volta), Some(_toolchain)) => (
+                Some(volta),
+                Some("this project is configured with both the deprecated `toolchain` key and the `volta` key; using the versions specified in `volta`.")
+            ),
+            (None, Some(toolchain)) => (Some(toolchain), Some("the `toolchain` key is deprecated and will be removed in a future version. Please switch to `volta` instead.")),
```

Comment:
Can we tell them the file (including path) to update? I realize its _probably_ always `package.json`, but IIRC we have some "find up" logic and it may be inferred from a path that isn't obvious.

Reply:
Oh, yes, that's smart. I'll see if there's a way to do that.